File: Advanced Methods of Programming/Tema3JAVA/src/main/java/validatori/ValidatorParticipant.java
package validatori;

import domain.Participant;

public class ValidatorParticipant implements Validator<Participant> {
    String err="";

    @Override
    public void validate(Participant entity) throws ValidationException {
        if(entity.getNumeParticipant()=="")
            err+="Numele participantului nu poate fi vid!\n";
        if(entity.getVarstaParticipant()<1||entity.getVarstaParticipant()>120)
            err+="Varsta incorecta!\n";

        if(err.length()>0)
            throw new ValidationException(err);
    }
}

File: Advanced Methods of Programming/Client-ServerCSharp/SharedDLL/Validators/ValidatorParticipant.cs
using SharedDLL.Domain;

namespace SharedDLL.Validators
{
    public class ValidatorParticipant : IValidator<Participant>
    {
        string err = "";
        public void Validate(Participant entity)
        {
            if(entity.GetNumeParticipant()=="")
                err += "Numele participantului nu poate fi vid!\n";
            if (entity.GetVarstaParticipant() < 1 || entity.GetVarstaParticipant() > 120)
                err += "Varsta incorecta!\n";

            if (err.Length > 0)
                throw new ValidationException(err);
        }
    }
}
